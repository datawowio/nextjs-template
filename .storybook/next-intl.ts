import en from "../src/locales/en/en.json";
import th from "../src/locales/th/th.json";

// noinspection ES6PreferShortImport
import { locales } from "../src/config/i18n";

const messagesByLocale: Record<string, unknown> = {
  en,
  th,
};

const nextIntl = {
  defaultLocale: locales[0],
  messagesByLocale,
};

export default nextIntl;
