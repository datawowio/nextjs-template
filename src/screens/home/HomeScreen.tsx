"use client";

import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import Stack from "@mui/material/Stack";

import { useTranslations } from "next-intl";
import Image from "next/image";

import logo from "@/assets/images/brands/logo.svg";
import Button from "@/components/core/button";
import Input from "@/components/core/input";
import Typography from "@/components/core/typography";
import { useToast } from "@/hooks/use-toast";

import { card, imageWrapper, root } from "./styles";

export default function HomeScreen() {
  const t = useTranslations("screens.home");

  const { showToast } = useToast();

  const handleClick = () => {
    showToast("This is a success message!", "success");
  };

  return (
    <Box sx={root}>
      <Card sx={card}>
        <Stack gap={4}>
          <Box sx={imageWrapper}>
            <Image src={logo} alt={t("logoAlt")} fill />
          </Box>
          <Typography customVariant="regularDisplayLG">
            {t("heading")}
          </Typography>
          <Typography>{t("description")}</Typography>
          <Button variant="contained" onClick={handleClick}>
            {t("cta")}
          </Button>
          <Input label="Enter your name" />
        </Stack>
      </Card>
    </Box>
  );
}
