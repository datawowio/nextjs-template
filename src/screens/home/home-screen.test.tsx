import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { useTranslations } from "next-intl";

import HomeScreen from "./home-screen";

jest.mock("next-intl", () => ({
  useTranslations: jest.fn(),
}));

describe("HomeScreen Component", () => {
  beforeEach(() => {
    (useTranslations as jest.Mock).mockImplementation(() => (key: string) => {
      const translations: Record<string, string> = {
        logoAlt: "Logo",
        heading: "Welcome to Home Screen",
        description: "This is the home screen of the application.",
        cta: "Get Started",
      };
      return translations[key];
    });
  });

  it("renders without crashing", () => {
    const { container } = render(<HomeScreen />);

    const logoImage = screen.getByAltText("Logo");
    const heading = screen.getByRole("heading", { level: 3 });
    const description = screen.getByText(
      "This is the home screen of the application.",
    );
    const button = screen.getByRole("button", { name: "Get Started" });

    expect(logoImage).toBeInTheDocument();
    expect(heading).toHaveTextContent("Welcome to Home Screen");
    expect(description).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
