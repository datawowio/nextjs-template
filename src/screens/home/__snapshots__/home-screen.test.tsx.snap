// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component renders without crashing 1`] = `
<div>
  <div
    class="MuiBox-root css-142ydkc"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1isd8c6-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiStack-root css-1ed8rzx-MuiStack-root"
      >
        <div
          class="MuiBox-root css-64lbpo"
        >
          <img
            alt="Logo"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Welcome to Home Screen
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          This is the home screen of the application.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Get Started
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
